catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -486,19 +496,36 @@ private void setResourceUriForContent(/*mutable*/ Map<String, Metacard> versionM
     private StorageProvider storageProvider() {
         return storageProviders.stream()
                 .findFirst()
-                .orElse(null);
+                .orElseThrow(() -> new RuntimeException(
+                        "Cannot version metacards without a storage provider"));
     }
 
     private CatalogProvider catalogProvider() {
         return catalogProviders.stream()
                 .findFirst()
-                .orElse(null);
+                .orElseThrow(() -> new RuntimeException(
+                        "Cannot version metacards without a storage provider"));
     }
 
     public void setMetacardTypes(List<MetacardType> metacardTypes) {
         this.metacardTypes = metacardTypes;
     }
 
+    private Predicate<Metacard> isVersionOrDeleted() {
+        return (m) -> {
+            String metacardTypeName = m.getMetacardType()
+                    .getName();
+
+            String metacardVersionTypeName = MetacardVersionImpl.getMetacardVersionType()
+                    .getName();
+            String deletedMetacardTypeName = DeletedMetacardImpl.getDeletedMetacardType()
+                    .getName();
+
+            return metacardVersionTypeName.equals(metacardTypeName)
```

Comment:
❓ Any way we could add a method to `MetacardVersionImpl` and `DeletedMetacardImpl` that would take in a `Metacard` or `MetacardType` and tell whether the `Metacard` is a version or a deleted one? It would make this code simpler and the logic reusable.